sycl/doc/extensions/proposed/sycl_ext_oneapi_enqueue_functions.asciidoc

```diff
@@ -0,0 +1,650 @@
+
+}
+----
+
+[source,c++]
+----
+template <typename KernelName, typename ExecutorType, typename LaunchConfig, typename KernelType>
+void parallel_for(ExecutorType ex, LaunchConfig lc, const KernelType& kernelFunc);
+----
+_Constraints_: Available only if `ExecutorType` is `sycl::handler`,
+`LaunchConfig` is `launch_config`, and the launch configuration was constructed
+from a `sycl::range`.
+
+_Effects_: Enqueues a basic kernel to the executor `ex` with the specified
+launch configuration.
+
+
+[source,c++]
+----
+template <typename KernelName, typename ExecutorType, typename LaunchConfig, typename KernelType>
+void parallel_for(ExecutorType ex, LaunchConfig lc, const KernelType& kernelFunc, const std::vector<event>& depEvents = {}, event* outEvent = nullptr);
+----
+_Constraints_: Available only if `ExecutorType` is `sycl::queue`,
+`LaunchConfig` is `launch_config`, and the launch configuration was constructed
+from a `sycl::range`.
+
+_Effects_: Enqueues a basic kernel to the executor `ex` with the specified
+launch configuration. The command will not execute until each of the events in
+`depEvents` is complete. An optional event representing this command can be
+returned via `outEvent`.
+
+
+=== ND-range kernels
+
+[source,c++]
+----
+namespace sycl::ext::oneapi::experimental {
+
+template <typename KernelName, typename ExecutorType, typename LaunchConfig, typename KernelType>
+void nd_launch(ExecutorType ex, LaunchConfig lc, const KernelType& kernelFunc);
+
+template <typename KernelName, typename ExecutorType, typename LaunchConfig, typename KernelType>
+void nd_launch(ExecutorType ex, LaunchConfig lc, const KernelType& kernelFunc, const std::vector<event>& depEvents = {}, event* outEvent = nullptr);
+
+}
+----
+
+[source,c++]
+----
+template <typename KernelName, typename ExecutorType, typename LaunchConfig, typename KernelType>
+void nd_launch(ExecutorType ex, LaunchConfig lc, const KernelType& kernelFunc);
+----
+_Constraints_: Available only if `ExecutorType` is `sycl::handler`,
+`LaunchConfig` is `launch_config`, and the launch configuration was constructed
+from a `sycl::nd_range`.
+
+_Effects_: Enqueues an ND-range kernel to the executor `ex` with the specified
+launch configuration.
+
+
+[source,c++]
+----
+template <typename KernelName, typename ExecutorType, typename LaunchConfig, typename KernelType>
+void nd_launch(ExecutorType ex, LaunchConfig lc, const KernelType& kernelFunc, const std::vector<event>& depEvents = {}, event* outEvent = nullptr);
+----
+_Constraints_: Available only if `ExecutorType` is `sycl::queue`,
+`LaunchConfig` is `launch_config`, and the launch configuration was constructed
+from a `sycl::nd_range`.
+
+_Effects_: Enqueues an ND-range kernel to the executor `ex` with the specified
+launch configuration. The command will not execute until each of the events in
+`depEvents` is complete. An optional event representing this command can be
+returned via `outEvent`.
+
```

Comment:
What about the versions of `single_task`, `parallel_for`, and `nd_launch` that take a `kernel` object?  We could just add this to the Issues section for now if you want.

Reply:
I've added an issue for this in 55caba3e9bfd, along with my thinking regarding potential future directions.